.github/scripts/automergeCrowdinPr.js

```diff
@@ -38,41 +49,46 @@ module.exports = async ({ github, context }) => {
     return
   }
 
-  console.log(`Verifying that only expected files are modified for PR #${pr.number}`)
-  const listFiles = await github.rest.pulls.listFiles({
+  console.log(`Fetching reviews for ${pr.number}`)
+  const listReviews = await github.rest.pulls.listReviews({
     owner,
     repo,
     pull_number: pr.number,
   })
-  const unexpectedFiles = listFiles.data.filter(
-    ({ filename }) =>
-      filename === DISALLOWED_UPDATED_FILE ||
-      !filename.match(new RegExp(ALLOWED_UPDATED_FILE_MATCHER))
-  )
-  if (unexpectedFiles.length > 0) {
-    console.log(
-      `Files updated in PR #${pr.number} do not match the expectation, please check manually`
-    )
-    await github.rest.pulls.createReview({
+  const isRejected = listReviews.data.some((review) => review.state === 'REQUEST_CHANGES')
+  const isApproved = listReviews.data.some((review) => review.state === 'APPROVED')
+
+  if (isRejected) {
+    console.log('Changes requested for this PR already, ending workflow')
+    return
+  } else if (isApproved) {
+    console.log('Already approved')
+  } else {
+    console.log(`Verifying that only expected files are modified for PR #${pr.number}`)
+    const listFiles = await github.rest.pulls.listFiles({
       owner,
       repo,
       pull_number: pr.number,
-      event: 'REQUEST_CHANGES',
-      body: `Changes requested from [${context.workflow} #${context.runNumber}](${context.serverUrl}/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}), as the updated files in this PR did not match the expectation. Please check.`,
     })
-    return
-  }
+    const unexpectedFiles = listFiles.data.filter(
+      ({ filename }) =>
+        filename === DISALLOWED_UPDATED_FILE ||
+        !filename.match(new RegExp(ALLOWED_UPDATED_FILE_MATCHER))
```

Comment:
super nit: this regexp could be reused instead of instantiating it for every file in the list (which is what's currently happening)

Reply:
(to reuse, pull it out of the filter callback)

Reply:
yeah good one, thanks! i've updated